File: benchmarks/poly/fibonacci/fibonacci.rb
def fib(n)
  return n if n < 2
  fib(n - 1) + fib(n - 2)
end

def main
  _t0 = Process.clock_gettime(Process::CLOCK_MONOTONIC, :nanosecond)
  (0..9).each do |i|
    puts fib(i)
  end
  $stderr.puts "bench_ns=#{Process.clock_gettime(Process::CLOCK_MONOTONIC, :nanosecond) - _t0}"
end

main

File: benchmarks/poly/fibonacci/fibonacci.pl
use Time::HiRes qw(clock_gettime CLOCK_MONOTONIC);

sub fib {
  my ($n) = @_;
  return $n if $n < 2;
  return fib($n - 1) + fib($n - 2);
}

sub main {
  my $_t0 = clock_gettime(CLOCK_MONOTONIC);
  for my $i (0 .. 9) {
    print fib($i) . "\n";
  }
  printf STDERR "bench_ns=%.0f\n", (clock_gettime(CLOCK_MONOTONIC) - $_t0) * 1e9;
}

main();
